hudi-cli/src/test/java/org/apache/hudi/cli/commands/TestLockAuditingCommand.java

```diff
@@ -282,4 +421,544 @@ public void testEnableLockAuditMultipleTimes() throws Exception {
+        () -> assertNotNull(result.toString()),
+        () -> assertTrue(result.toString().contains("Validation Result: PASSED")),
+        () -> assertTrue(result.toString().contains("Transactions Validated: 2")),
+        () -> assertTrue(result.toString().contains("Issues Found: 0")),
+        () -> assertTrue(result.toString().contains("successfully")));
+  }
+
+  /**
+   * Test validation - Single Unclosed Transaction (WARNING)
+   */
+  @Test
+  public void testValidateAuditLocksSingleUnclosedTransaction() throws IOException {
+    long baseTime = 1000000L;
+    List<TransactionScenario> scenarios = new ArrayList<>();
+    
+    // Single audit file without END record
+    List<AuditRecord> records1 = new ArrayList<>();
+    records1.add(new AuditRecord("owner1", baseTime, baseTime + 100, "START", baseTime + 200));
+    scenarios.add(new TransactionScenario(baseTime + "_owner1.jsonl", records1));
+    
+    createAuditFiles(scenarios);
+    
+    Object result = shell.evaluate(() -> "locks audit validate");
+    
+    assertAll("Single unclosed transaction shows warning",
+        () -> assertTrue(ShellEvaluationResultUtil.isSuccess(result)),
+        () -> assertNotNull(result.toString()),
+        () -> assertTrue(result.toString().contains("Validation Result: WARNING")),
+        () -> assertTrue(result.toString().contains("Transactions Validated: 1")),
+        () -> assertTrue(result.toString().contains("Issues Found: 1")));
+  }
+
+  /**
+   * Test validation - Unclosed Transactions (WARNING)
+   */
+  @Test
+  public void testValidateAuditLocksUnclosedTransactions() throws IOException {
+    long baseTime = 1000000L;
+    List<TransactionScenario> scenarios = new ArrayList<>();
+    
+    // Transaction 1: Unclosed (effective end at expiration = baseTime + 200)
+    List<AuditRecord> records1 = new ArrayList<>();
+    records1.add(new AuditRecord("owner1", baseTime, baseTime + 100, "START", baseTime + 200));
+    scenarios.add(new TransactionScenario(baseTime + "_owner1.jsonl", records1));
+    
+    // Transaction 2: Complete, starts after owner1's expiration
+    List<AuditRecord> records2 = new ArrayList<>();
+    records2.add(new AuditRecord("owner2", baseTime + 300, baseTime + 400, "START", baseTime + 60000));
+    records2.add(new AuditRecord("owner2", baseTime + 300, baseTime + 500, "END", baseTime + 60000));
+    scenarios.add(new TransactionScenario((baseTime + 300) + "_owner2.jsonl", records2));
+    
+    createAuditFiles(scenarios);
+    
+    Object result = shell.evaluate(() -> "locks audit validate");
+    
+    assertAll("Unclosed transactions show warning",
+        () -> assertTrue(ShellEvaluationResultUtil.isSuccess(result)),
+        () -> assertNotNull(result.toString()),
+        () -> assertTrue(result.toString().contains("Validation Result: WARNING")),
+        () -> assertTrue(result.toString().contains("Transactions Validated: 2")),
+        () -> assertTrue(result.toString().contains("Issues Found: 1")),
```

Comment:
Is unclosed transactions expected?  Would `WARNING` be noisy in this case?

Reply:
yes, no it would not be noisy